api/v1alpha1/bmctask_types.go

```diff
@@ -50,7 +58,7 @@ type Task struct {
 
 type PowerAction struct {
 	// State represents the requested power state to set for the baseboard management.
-	// +kubebuilder:validation:Enum=PowerOn;HardPowerOff;SoftPowerOff;Status;Cycle;Reset
+	// +kubebuilder:validation:Enum=on;off;soft;status;cycle;reset
 	PowerControl PowerControl `json:"powerControl"`
 }
```

Comment:
Can this just be a `PowerAction` on the task rather than having the struct?

```go
type PowerAction string

const (
	PowerActionOn PowerAction "PowerOn"
	PowerActionHardPowerOff PowerAction "HardPowerOff"
)

type Task struct {
	// PowerAction represents a baseboard management power operation.
	PowerAction *PowerAction `json:"powerAction,omitempty"`
	// OneTimeBootDeviceAction represents a baseboard management one time set boot device operation.
	OneTimeBootDeviceAction *OneTimeBootDeviceAction `json:"oneTimeBootDeviceAction,omitempty"`
}
```